lib/network.go

```diff
@@ -854,23 +854,6 @@ func (msg *MsgDeSoDisconnectedPeer) FromBytes(data []byte) error {
 	return fmt.Errorf("MsgDeSoDisconnectedPeer.FromBytes not implemented")
 }
 
-// MsgDeSoPeerHandshakeComplete is a control message that is used to internally signal when a peer has
```

Comment:
Why remove this control message type?

Reply:
Yeah, this is a fun simplification. The `MsgDeSoPeerHandshakeComplete` message turned out to be redundant: it was emitted by the HandshakeManager, and then handled back at HandshakeManager. Not only was this an unnecessary redirection, it actually caused deadlocking! 

The thread that was enqueuing the `HandshakeComplete` message to the server's `incomingMessages` channel was the same thread that was dequeueing messages from the `incomingMessages`. So if the `incomingMessages` channel was full as we're enqueueing the `HandshakeComplete` message, we would have deadlocked, since no thread would be able to evict the `incomingMessages` channel.

The simplest solution was to just remove the `HandshakeComplete` message indirection, and replace it with a direct call to HandshakeManager's handler for the `HandshakeComplete` message.